packages/eas-cli/src/build/utils/resourceClass.ts

```diff
@@ -0,0 +1,135 @@
+  [ResourceClass.M1_LARGE]: BuildResourceClass.IosM1Large,
+  [ResourceClass.INTEL_MEDIUM]: BuildResourceClass.IosIntelMedium,
+  [ResourceClass.MEDIUM]: BuildResourceClass.IosMedium,
+};
+
+const androidResourceClassToBuildResourceClassMapping: Record<
+  Exclude<
+    ResourceClass,
+    | ResourceClass.M1_EXPERIMENTAL
+    | ResourceClass.M1_MEDIUM
+    | ResourceClass.M1_LARGE
+    | ResourceClass.INTEL_MEDIUM
+  >,
+  BuildResourceClass
+> = {
+  [ResourceClass.DEFAULT]: BuildResourceClass.AndroidDefault,
+  [ResourceClass.LARGE]: BuildResourceClass.AndroidLarge,
+  [ResourceClass.MEDIUM]: BuildResourceClass.AndroidMedium,
+};
+
+async function resolveIosDefaultRequestedResourceClassAsync(
+  getDynamicProjectConfigAsync: DynamicConfigContextFn,
+  projectDir: string,
+  buildProfile: ProfileData<'build'>
+): Promise<ResourceClass> {
+  const { exp } = await getDynamicProjectConfigAsync({ env: buildProfile.profile.env });
+  const { sdkVersion } = exp;
+  const reactNativeVersion = await getReactNativeVersionAsync(projectDir);
+  if (
+    (sdkVersion && semver.satisfies(sdkVersion, '>=48')) ||
+    (reactNativeVersion && semver.satisfies(reactNativeVersion, '>=0.71.0'))
+  ) {
+    return ResourceClass.M1_MEDIUM;
+  } else {
+    return ResourceClass.DEFAULT;
+  }
+}
+
+function resolveAndroidResourceClass(
+  profileResourceClass?: ResourceClass,
+  resourceClassFlag?: ResourceClass
+): BuildResourceClass {
+  if (
+    resourceClassFlag &&
+    [
+      ResourceClass.M1_EXPERIMENTAL,
+      ResourceClass.M1_MEDIUM,
+      ResourceClass.M1_LARGE,
+      ResourceClass.INTEL_MEDIUM,
+    ].includes(resourceClassFlag)
+  ) {
+    throw new Error(`Resource class ${resourceClassFlag} is only available for iOS builds`);
+  }
+
+  const resourceClass = resourceClassFlag ?? profileResourceClass ?? ResourceClass.DEFAULT;
+
+  return androidResourceClassToBuildResourceClassMapping[
+    resourceClass as Exclude<
+      ResourceClass,
+      | ResourceClass.M1_EXPERIMENTAL
+      | ResourceClass.M1_MEDIUM
+      | ResourceClass.M1_LARGE
+      | ResourceClass.INTEL_MEDIUM
+    >
+  ];
+}
+
+async function resolveIosResourceClassAsync(
+  buildProfile: ProfileData<'build'>,
+  getDynamicProjectConfigAsync: DynamicConfigContextFn,
+  projectDir: string,
+  profileResourceClass?: ResourceClass,
+  resourceClassFlag?: ResourceClass
+): Promise<BuildResourceClass> {
+  const resourceClass =
+    resourceClassFlag ??
+    profileResourceClass ??
+    (await resolveIosDefaultRequestedResourceClassAsync(
+      getDynamicProjectConfigAsync,
+      projectDir,
+      buildProfile
+    ));
+
+  if (resourceClass === ResourceClass.M1_EXPERIMENTAL) {
+    Log.warn(`Resource class ${chalk.bold('m1-experimental')} is deprecated.`);
+  }
+
+  if ([ResourceClass.LARGE, ResourceClass.M1_LARGE].includes(resourceClass)) {
+    Log.warn(
+      `Large resource classes are not available for iOS builds yet. Your build will use the medium resource class.`
+    );
```

Comment:
When those values are specified we still send large to www and display large on website. So even now technically this is not true.
Also this message will also show up after we introduce large classes which will be even more confusing

Reply:
>When those values are specified we still send large to www and display large on website. So even now technically this is not true.

This is not true, we don't display `large` on the website in such case, since we changed how the `resourceClass` resolver works https://github.com/expo/universe/blob/main/server/www/src/graphql/types/Build.ts#L317-L318
I also tested it, this is the build submitted with `large` resource class: https://expo.dev/accounts/szdziedzic/projects/testapp/builds/9560591e-b00a-4d92-9f43-ca77545288e5

>Also this message will also show up after we introduce large classes which will be even more confusing

That's right. Do you suggest dropping it? It might be confusing to users who use the deprecated flag with a `large` resource class. They will start the build with the `large` resource class and see `Intel Medium` on the website.

Reply:
> Do you suggest dropping it?

yes, there is no large class for iOS in documentation, so it's on users if it's not used correctly, I believe we should think first about the experience of users that use supported stuff and add warnings like that only it does not affect those users.

Reply:
Sounds reasonable, thanks for this discussion!